sys/kern/src/arch/riscv32.rs

```diff
@@ -607,11 +604,9 @@ unsafe fn handle_fault(task: *mut task::Task, fault: FaultInfo) {
             }
 
             let next = &mut tasks[next];
-            apply_memory_protection(next);
-            // Safety: this leaks a pointer aliasing into static scope, but
-            // we're not going to read it back until the next kernel entry so
-            // we won't be aliasing/racing.
-            set_current_task(next);
+            // Safety: next comes from teh task table and we don't use it again
```

Comment:
Not sure how much you care about fixing typos. 'teh' is in here in a few places from the existing code. however, I suspect the diff being generated more prominently shows it as part of the diff proper.

Reply:
Good point. Let me fix that in a follow-up commit.

Reply:
Done. Pushed a follow up change.